stripzip.py

```diff
@@ -10,34 +10,104 @@ class NonZipFileError(ValueError):
 
 
 def _zero_zip_date_time(zip_):
-    archive_size = os.fstat(zip_.fileno()).st_size
+    def purify_extra_data(mm, offset, length):
+        extra_header_struct = Struct("<HH")
+        # 0. id
+        # 1. length
+        STRIPZIP_OPTION_HEADER = 0xFFFF
+        EXTENDED_TIME_DATA = 0x5455
+        # Some sort of extended time data, see
+        # ftp://ftp.info-zip.org/pub/infozip/src/zip30.zip ./proginfo/extrafld.txt
+        # fallthrough
+        UNIX_EXTRA_DATA = 0x7875
+        # Unix extra data; UID / GID stuff, see
+        # ftp://ftp.info-zip.org/pub/infozip/src/zip30.zip ./proginfo/extrafld.txt
+        mlen = offset + length
+
+        while offset < mlen:
+            values = list(extra_header_struct.unpack_from(mm, offset))
+            _, header_length = values
+            extra_struct = Struct("<HH" + "B"*header_length)
+            values = list(extra_struct.unpack_from(mm, offset))
+            header_id, header_length, rest = values[0], values[1], values[2:]
+
+            if header_id in (EXTENDED_TIME_DATA, UNIX_EXTRA_DATA):
+                values[0] = STRIPZIP_OPTION_HEADER
+                for i in xrange(2, len(values)):
+                    values[i] = 0xff
+                extra_struct.pack_into(mm, offset, *values)
+            elif header_id != STRIPZIP_OPTION_HEADER:
+                return False
+ 
+            offset += extra_header_struct.size + header_length
+
+        return True
```

Comment:
https://github.com/KittyHawkCorp/stripzip/blob/master/src/stripzip_app.c#L118 so the `purify_extra_data` is  a backport of that function implemented in C, this is the reason for returning true/false, the return is ignored even in the original implementation, so not really an issue. We can raise an error in case, but I don't find it necessary.

Reply:
https://github.com/KittyHawkCorp/stripzip/blob/master/src/stripzip_app.c#L118 so the `purify_extra_data` is  a backport of that function implemented in C, this is the reason for returning true/false, the return is ignored even in the original implementation, so not really an issue. We can raise an error in case, but I don't find it necessary.